### PR TITLE
Improve SDS server to support multiple UDS

### PR DESCRIPTION
**Pull request by wenchenglu**

### CODE REVIEW

File: security/pkg/workload/server.go [hunk 1]
```diff
@@ -103,25 +116,29 @@ func (s *SDSServer) StreamSecrets(stream sds.SecretDiscoveryService_StreamSecret
...
+}
 
+// RegisterUdsPath registers a path for Unix Domain Socket and has
+// SDSServer's gRPC server listen on it.
+func (s *SDSServer) RegisterUdsPath(udsPath string) error {
+	s.udsPaths = append(s.udsPaths, udsPath)
 	_, err := os.Stat(udsPath)
 	if err == nil {
-		return nil, fmt.Errorf("UDS path %v already exists", udsPath)
+		return fmt.Errorf("UDS path %v already exists", udsPath)
```
Comment (wattli) [inline security/pkg/workload/server.go]: Please use log.Errorf() as above.

File: security/pkg/workload/server.go [hunk 2]
```diff
@@ -69,11 +73,20 @@ func (s *SDSServer) GetTLSCertificate(udsPath string) *auth.TlsCertificate {
 // FetchSecrets fetches the X.509 key/cert for a given workload whose identity
 // can be derived from the UDS path where this call is received.
 func (s *SDSServer) FetchSecrets(ctx context.Context, request *api.DiscoveryRequest) (*api.DiscoveryResponse, error) {
+	// Get the uds path where this call is received.
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok || len(md[udsPathKey]) == 0 {
+		errMessage := "Unknown UDS path."
```
Comment (wattli) [inline security/pkg/workload/server.go]: s/Unknown/unknown 

Typically the error message starts with the first word in small case.

Comment (wenchenglu) [reply security/pkg/workload/server.go]: done

File: security/pkg/workload/server.go (hunk 1 above)
Comment (wenchenglu) [reply security/pkg/workload/server.go]: This is for returning the error string which will be logged by the caller. I thought log.Errorf() is to print it to the error log.

**Pull request merged**